question26/js/login.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const submitButton = document.getElementById('js-submit-button');
+const userInfoInputArea = document.getElementById('name_mail');
+const passwordInputArea = document.getElementById('password');
+const form = document.getElementById('form');
+
+if(localStorage.hasOwnProperty('token')) {
+  window.location.href = './newsComponents/index.html'
+}
+
+const isValidStatus = {
+  name_mail: false,
+  password: false,
+}
+
+const validationInfo = {
+  name_mail: {
+    validation: (value) => value
+  },
+  password: {
+    validation : (value) => /(?=.*[a-zA-Z])(?=.*[0-9])[a-zA-z0-9]{8,}/.test(value),
+    errorMessage: '※8文字以上の大小の英数字を交ぜたものにしてください。'  
+  }
+}
+
+const setValidationEvents = (isValid, targetForm) => {
+  if (isValid) {
+    targetForm.classList.add('valid');
+    targetForm.classList.remove('invalid');
+    targetForm.nextElementSibling.textContent = ''; 
+    isValidStatus[targetForm.id] = true;
+  } else {
+    targetForm.classList.add('invalid');
+    targetForm.classList.remove('valid');
+    targetForm.nextElementSibling.textContent = validationInfo[targetForm.id].errorMessage;
+    isValidStatus[targetForm.id] = false; 
+  }
+}
+
+const validateInputValue = (e) => {
+  const targetForm = e.target;
+  const value = targetForm.value.trim();
+  const result = validationInfo[targetForm.id].validation(value);
+  setValidationEvents(result,targetForm);
+  renderRequiredFieldMessages(targetForm);  
+  switchSubmitButton(checkAllValidity());
+} 
+
+const checkAllValidity = () => {
+  return Object.values(isValidStatus).every((result) => result );
+}
+
+const switchSubmitButton = (isValid) => {
+  submitButton.disabled = isValid ? false : true;
+} 
```

Comment:
IMO 
I thought it would be better to use a function name that is easy to understand that it returns a boolean value.

```suggestion
const isDisabledSubmitButton = (isValid) => {
  submitButton.disabled = isValid ? false : true;
} 
```

Reply:
Thank you for your review!
It is true! I have made the changes as per your suggestion. Please confirm.

fixed commit : https://github.com/csakyo/homework/pull/40/commits/af913642abc18c3ef82989848b98b26014fa8611

Reply:
横からすみません…

> that it returns a boolean value.
ここはBooleanを返していないと思うので
is**ではなく、disabledをtoggleしていることがわかる関数名の方が適切に感じます。

Reply:
@sae-github 
ご指摘ありがとうございます！！
checkedの値を、切り替えている関数ですね。。
失礼しました。

@csakyo 
間違ったことをお伝えして、大変申し訳ありません。
さえさんのいう通り、切り替えていることをわかる関数名が適切だと思います。
switchという名前はあまり使わないそうなので、違う名前がいいかもしれません。
https://github.com/seeeen044/JavaScript_handson/pull/19#discussion_r886920139

Reply:
@sae-github @nakagawamai 
ご指摘いただきましてありがとうございます:bow: 自分も見落としておりました..すみません。。
ボタンの無効化の切り替えという意味で修正してみましたがいかがでしょうか？ご確認お願いいたします。

fixed commit : https://github.com/csakyo/homework/pull/40/commits/4e3c8c14668228f5c56d8398524292fb24c69bde

Reply:
I felt it would be easier to understand if I put "disabled" in the function name since it is a switch for "disabled"!

ex.)
``` js
toggleDisabledOfSubmitButton
```

Reply:
Oh, I see...! I've made the corrections as suggested :bow: Please confirm.

fixed commit : https://github.com/csakyo/homework/pull/40/commits/01dc123d1dbb909e38af2185cbd71ebf49ba22ca